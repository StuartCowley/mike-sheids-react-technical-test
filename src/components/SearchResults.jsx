import React from "react";
import { v4 as uuidv4 } from "uuid";
import PropTypes from "prop-types";
import "../styles/search-results.css";

const SearchResults = ({ results }) => {
  if (!results.length) {
    return <p>No results</p>;
  }

  return (
    <div className="search-results" data-testid="search-results">
      {results.map((url) => (
        <img
          key={uuidv4()}
          className="search-results__image"
          src={url}
          alt=""
        />
      ))}
    </div>
  );
};

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default SearchResults;
