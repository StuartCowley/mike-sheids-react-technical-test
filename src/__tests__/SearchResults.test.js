import React from "react";
import { render, screen } from "@testing-library/react";
import mockApiResponse from "./mockApiResponse.json";
import SearchResults from "../components/SearchResults";

const validProps = {
  results: mockApiResponse.collection.items,
};
const renderSearchResults = () => render(<SearchResults {...validProps} />);

describe("SearchResults", () => {
  it("matches snapshot", () => {
    const { asFragment } = renderSearchResults();

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays a collection of image elements", () => {
    renderSearchResults();

    mockApiResponse.collection.items.forEach((item) => {
      const { title } = item.data[0]

      expect(screen.getByAltText(title)).toBeInTheDocument();
    });
  });
});
