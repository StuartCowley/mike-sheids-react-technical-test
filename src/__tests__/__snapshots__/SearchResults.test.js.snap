// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="search-results"
    data-testid="search-results"
  >
    <div>
      <div
        style="display: flex; flex-direction: row; justify-content: center; align-content: stretch; box-sizing: border-box; width: 100%; gap: 1rem;"
      >
        <div
          style="display: flex; flex-direction: column; justify-content: flex-start; align-content: stretch; flex: 1; width: 0px; gap: 1rem;"
        >
          <img
            alt=""
            class="search-results__image"
            src="https://images-assets.nasa.gov/image/PIA12235/PIA12235~thumb.jpg"
          />
          <img
            alt=""
            class="search-results__image"
            src="https://images-assets.nasa.gov/image/PIA12228/PIA12228~thumb.jpg"
          />
        </div>
        <div
          style="display: flex; flex-direction: column; justify-content: flex-start; align-content: stretch; flex: 1; width: 0px; gap: 1rem;"
        >
          <img
            alt=""
            class="search-results__image"
            src="https://images-assets.nasa.gov/image/PIA13517/PIA13517~thumb.jpg"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
